src/summary/history/allergy-edit-form.component.tsx

```diff
@@ -0,0 +1,330 @@
+import React from "react";
+import { match } from "react-router";
+import {
+  getPatientAllergyByPatientUuid,
+  getAllergyReaction,
+  updatePatientAllergy,
+  deletePatientAllergy
+} from "./allergy-intolerance.resource";
+import { useCurrentPatient } from "@openmrs/esm-api";
+import { createErrorHandler } from "@openmrs/esm-error-handling";
+import SummaryCard from "../cards/summary-card.component";
+import style from "./allergy-form.css";
+import dayjs, { Dayjs } from "dayjs";
+import { useHistory } from "react-router-dom";
+
+export function AllergyEditForm(props: AllergyEditFormProps) {
+  const history = useHistory();
+  const [patientAllergyReaction, setPatientAllergyReaction] = React.useState(
+    []
+  );
+  const [patientAllergy, setPatientAllergy] = React.useState(null);
+  const [allergyReaction, setAllergyReaction] = React.useState();
+  const [enableButtons, setEnableButtons] = React.useState(true);
+  const formRef = React.useRef<HTMLFormElement>(null);
+  const [reactionSeverityUuid, setReactionSeverityUuid] = React.useState(null);
+  const [allergyComment, setAllergyComment] = React.useState(null);
+  const [updatedDate, setUpdatedDate] = React.useState(null);
+
+  const [
+    isLoadingPatient,
+    patient,
+    patientUuid,
+    patientErr
+  ] = useCurrentPatient();
+
+  const setCheckedValue = uuid => {
+    return patientAllergy.reactions.some(
+      reaction => reaction.reaction.uuid === uuid
+    );
+  };
+
+  const handleAllergenReactionChange = event => {
+    if (event.target.checked == true) {
+      setPatientAllergyReaction([
+        ...patientAllergyReaction,
+        { uuid: event.target.value }
+      ]);
+    } else {
+      patientAllergyReaction.splice(
+        patientAllergyReaction.findIndex(
+          reaction => reaction.uuid == event.target.value
+        ),
+        1
+      );
+    }
+  };
+
+  const handleSubmit = event => {
+    event.preventDefault();
+    const allergy: Allergy = {
+      allergenType: patientAllergy.allergen.allergenType,
+      codedAllergenUuid: patientAllergy.allergen.codedAllergen.uuid,
+      severityUuid: reactionSeverityUuid,
+      comment: allergyComment,
+      reactionsUuid: patientAllergyReaction
+    };
+    const abortController = new AbortController();
+    updatePatientAllergy(
+      allergy,
+      patientUuid,
+      props.match.params,
+      abortController
+    ).then(response => {
+      response.status == 200 && navigate();
+    }, createErrorHandler);
+    return () => abortController.abort();
+  };
+
+  const handleDeletePatientAllergy = () => {
+    const abortController = new AbortController();
+    deletePatientAllergy(patientUuid, props.match.params, abortController).then(
+      response => {
+        response.status == 204 && navigate();
```

Comment:
why are you using && here?

Reply:
to navigate to level two only when the delete request is successful